librepo/checksum.h

```diff
@@ -107,6 +107,38 @@ lr_checksum_fd_compare(LrChecksumType type,
                        gchar **calculated,
                        GError **err);
 
+/** Calculate checksum for data pointed by file descriptor and
+ * compare it to the expected checksum value. It also sets the timestamp
+ * of the file to given value.
+ * @param type          Checksum type
+ * @param fd            File descriptor
+ * @param expected      String with expected checksum value
+ * @param caching       Cache/Use cached checksum value as extended file attr.
+ * @param matches       Set pointed variable to TRUE if checksum matches.
+ * @param calculated    If not NULL, the calculated checksum will be pointed
+ *                      here, the pointed string must be freed by caller.
+ * @param timestamp     Use this timestamp for checksum caching. If -1,
+ *                      timestamp is taken from file's mtime.
+ * @param err           GError **
+ * @return              returns TRUE if error is not set and FALSE if it is
+ */
+gboolean
+lr_checksum_fd_compare_mtime(LrChecksumType type,
+                       int fd,
```

Comment:
I see you are still updating this so I didn't look into it too much but can you fix the alignment of arguments for `lr_checksum_fd_compare_mtime` here and also its definition has the same problem.

Additionally I think the `test_cached_checksum` unit test in test_checksum.c needs an update.

Reply:
I fixed the indent issue and will look to the unit test stuff.

Reply:
It looks like the `test_cached_checksum` was never working well - the test file is created in `/tmp/` which is usually a tmpfs filesystem. Even though tmpfs supports extended attributes, usage of the `user` xattrs (on which librepo's checksum caching relies) is not permitted here. (see man tmpfs)

Reply:
I see, that would explain why the test was passing on my host machine (it was skipping the check because it was on tmpfs) but it was failing in podman.

Reply:
I fixed the `test_cached_checksum` test and added new  tests for `lr_checksum_fd_compare_mtime` and `lr_checksum_clear_cache`.
I'm not sure if i should try to create temp files out of the /tmp/ directory - even then you can never be sure that the filesystem you are using supports them.

Reply:
I agree, the current approach in the unittests is sufficient.

What about the CI tests? Will we have the same problem there? I tried searching for what is the status of xattrs in containers but didn't manage to find anything definitive. It does work in my podman runs.

Reply:
Or it actually shouldn't matter from reposync's point of view, it will just compute the checksum again, but the behavior would be the same?

Reply:
That's a good point actually. When the CI test runs on the tmpfs, the extended attributes are not used, the checksums are re-calculated and the re-downloads will not occur - the test will pass regardless the patch is present.
Although I do not have the CI test for the bug ready yet, from what I've seen the extended attributes are working in our test containers so I'll rely on that and prepare the test. I will add a step there checking that the extended attribute on the downloaded file is actually present - just to make it fail in case the xattrs are not supported.